pkg/oc/cli/newapp/newapp.go

```diff
@@ -1019,7 +1019,11 @@ func TransformRunError(err error, baseName, commandName, commandPath string, gro
 		// TODO: suggest things to the user
 		groups.Add("", "", "", UsageError(commandPath, newAppNoInput, baseName, commandName))
 	default:
-		groups.Add("", "", "", err)
+		if runtime.IsNotRegisteredError(err) {
+			groups.Add("", "", "", fmt.Errorf("The template contained an object type unknown to `oc new-app`.  Use `oc process | oc create` instead.  The decoder returned: %s", err.Error()))
```

Comment:
Full command to run as the work-around is `oc process -f mytemplate.json | oc create -f -`.

Not sure if we can capture the filename in the error message so we can produce a valid command.

Reply:
I'll investigate and report back

Reply:
am able to provide filename ... will update shortly